src/internal/connector/onedrive/item.go

```diff
@@ -344,3 +341,36 @@ func driveItemWriter(
 
 	return uploadsession.NewWriter(itemID, url, itemSize), nil
 }
+
+// constructWebURL helper function for recreating the webURL
+// for the originating SharePoint site. Uses additional data map
+// from a models.DriveItemable that possesses a downloadURL within the map.
+// Returns "" if map nil or key is not present.
+func constructWebURL(adtl map[string]any) string {
+	var (
+		desiredKey = "@microsoft.graph.downloadUrl"
+		del        = `/_layouts`
```

Comment:
nit: `del` reads like `delete`
```suggestion
		sep        = `/_layouts`
```

Reply:
can we be certain that the stuff we're working with will always have `_layouts` in it? This seems like a fragile part of the code that could break -- possibly silently -- without warning :/

Are there other options we have for retrieving this information? For example, is it available higher up the stack and can it be passed down?

Reply:
Great question! 

This is not the fundamental way web URLs are retrieved. This is an alternative. The primary way is found here:

https://github.com/alcionai/corso/blob/7b650ebe8de4641a67472615a6a56c3f68973355/src/internal/connector/onedrive/item.go#L288

With the other fix, the `sharepointIds`, the URL should not be empty. In the peculiar corner case that it is, the download URL possesses the information.

Reply:
hm, alright. We should keep an eye on this in the future though